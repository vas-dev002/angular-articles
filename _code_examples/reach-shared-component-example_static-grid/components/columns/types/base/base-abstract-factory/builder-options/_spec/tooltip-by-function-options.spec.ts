import { ITooltipParams } from "@ag-grid-community/core";
import { TooltipByFunctionOptions } from "..";

declare const expect: jest.Expect;

describe("[Option]: TooltipByFunctionOptions", () => {
	let tooltipByFunctionOptions: TooltipByFunctionOptions;

	beforeEach(() => {
		tooltipByFunctionOptions = new TooltipByFunctionOptions();
	});

	describe("[TooltipByFunctionOptions instantiate]", () => {
		it("should be successfully created", () => {
			expect(tooltipByFunctionOptions).toBeTruthy();
			expect(tooltipByFunctionOptions).toMatchSnapshot();
		});
	});

	describe("[Method]: tooltipValueGetter", () => {
		it("should return valid value when slot is invalid", () => {
			const params = { value: "Test1" } as ITooltipParams;
			const tooltipValueGetter = tooltipByFunctionOptions.tooltipValueGetter(params);

			expect(tooltipValueGetter).toBe(params.value);
		});
	});
});
