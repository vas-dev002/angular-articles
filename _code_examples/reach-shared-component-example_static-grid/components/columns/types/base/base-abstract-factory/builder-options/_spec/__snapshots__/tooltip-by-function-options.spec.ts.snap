// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Option]: TooltipByFunctionOptions [TooltipByFunctionOptions instantiate] should be successfully created 1`] = `
TooltipByFunctionOptions {
  "tooltipComponentFramework": [Function],
  "tooltipValueGetter": [Function],
}
`;
